numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -3423,31 +3431,57 @@ PyArray_Where(PyObject *condition, PyObject *x, PyObject *y)
         /* Get the result from the iterator object array */
         ret = (PyObject*)NpyIter_GetOperandArray(iter)[0];
 
-        NPY_BEGIN_THREADS_NDITER(iter);
+        npy_intp itemsize = common_dt->elsize;
+
+        int swap = PyDataType_ISBYTESWAPPED(common_dt);
+        int native = (swap == 0) && !needs_api;
```

Comment:
```suggestion
int has_ref = PyDataType_REFCHK(common_dt);
```
Oh lets just delete it, this is meaningless.  It can never be swapped.